### DISCUSSION TITLE
Obsidian test vault to help contributors hit the ground running

### DISCUSSION THREAD

**Discussion by blakeNaccarato** [anoopkcn/obsidian-reference-map#24] [Ideas]
While working on the latest citekey bug, as you know I set up https://github.com/blakeNaccarato/obsidian-reference-map-test. I've configured the repo now to track the latest commit of your plugin, with a dependency manager that will automatically bump the submodule. This repo could serve as a "good enough for now" test aid, where you can draft up some edge cases and see how your features are working.

But I think a similar template vault could facilitate contributions. You could put in `CONTRIBUTING.md` for users to clone a separate test vault repo (similar to the one above). Maybe it would have a `setup.ps1` or `setup.sh` script which takes the user's fork URL and desired PR branch name as arguments, and handles cloning, branch creation, `npm install` and `npm run dev`, etc. It would cut the barrier to entry for drive-by contributors down from a few hours (for total noobs like myself) to ~fifteen minutes, and they'd be able to open a test vault and start seeing their changes actually take effect in Obsidian.

If you are interested, and have input on how such a contribution helper repo would look, I could work on it on my free time and draft up the `CONTRIBUTING.md` details for it.

**Comment by anoopkcn** [comment]
Hi, Thanks for this. I will take a look at everything during the weekend.  It's hard to find free time during the working days :)
